Loop/View Controllers/StatusTableViewController.swift

```diff
@@ -2014,56 +2028,173 @@ extension StatusTableViewController: SettingsViewModelDelegate {
+        log.default("Service with identifier '%{public}@' created", service.serviceIdentifier)
+        deviceManager.servicesManager.addActiveService(service)
     }
 }
 
-extension StatusTableViewController: ServicesViewModelDelegate {
-    func addService(identifier: String) {
-        setupService(withIdentifier: identifier)
+extension StatusTableViewController: ServiceOnboardDelegate {
+    func serviceOnboardNotifying(_ notifying: ServiceOnboardNotifying, didOnboardService service: Service) {
+        precondition(service.isOnboarded)
+        log.default("Service with identifier '%{public}@' onboarded", service.serviceIdentifier)
     }
-    func gotoService(identifier: String) {
-        guard let serviceUI = deviceManager.servicesManager.activeServices.first(where: { $0.serviceIdentifier == identifier }) as? ServiceUI else {
+}
+
+// MARK: - Onboarding
+
+extension StatusTableViewController {
+    fileprivate func setupOnboarding(withIdentifier identifier: String) {
+        guard let onboardingUIType = deviceManager.pluginManager.getOnboardingTypeByIdentifier(identifier) else {
             return
         }
-        showServiceSettings(serviceUI)
+
+        let onboarding = onboardingUIType.init()
+        var onboardingViewController = onboarding.onboardingViewController(preferredGlucoseUnit: deviceManager.preferredGlucoseUnit,
+                                                                           cgmManagerProvider: self,
+                                                                           pumpManagerProvider: self,
+                                                                           serviceProvider: self,
+                                                                           colorPalette: .default)
+        onboardingViewController.onboardingDelegate = self
+        onboardingViewController.cgmManagerCreateDelegate = self
+        onboardingViewController.cgmManagerOnboardDelegate = self
+        onboardingViewController.pumpManagerCreateDelegate = self
+        onboardingViewController.pumpManagerOnboardDelegate = self
+        onboardingViewController.serviceCreateDelegate = self
+        onboardingViewController.serviceOnboardDelegate = self
+        onboardingViewController.completionDelegate = self
+
+        show(onboardingViewController, sender: self)
     }
+}
 
-    fileprivate func showServiceSettings(_ serviceUI: ServiceUI) {
-        var settings = serviceUI.settingsViewController(currentTherapySettings: deviceManager.loopManager.therapySettings, preferredGlucoseUnit: deviceManager.preferredGlucoseUnit, chartColors: .primary, carbTintColor: .carbTintColor, glucoseTintColor: .glucoseTintColor, guidanceColors: .default, insulinTintColor: .insulinTintColor)
-        settings.serviceSettingsDelegate = self
-        settings.completionDelegate = self
-        show(settings, sender: self)
+struct UnknownIdentifierError: Error {}
+
+extension StatusTableViewController: CGMManagerProvider {
```

Comment:
[nit] Is it time to split this file up...or at least pull out some of these `extension` protocol conformances?

Reply:
IMO, now is not the time. The appropriate time for that work will be when we get to https://tidepool.atlassian.net/browse/LOOP-2546.